### ISSUE TITLE
LRG flux = 0 for some tests

### ISSUE THREAD

**Issue by sbailey** [desihub/desisim#195]
We have a random test failure where an LRG template is returned with all zeros.  This is reproducible when using desisim-testdata/master and the following code:
```shell
$> source desisim-testdata/setup-testdata.sh
$> cd desisim
$> ipython
```
from python:
```
from desisim.templates import LRG
lrg = LRG()
simflux, wave1, meta = lrg.make_templates(nmodel=2, seed=651684709)
print(simflux)  #- all zeros (!)
```
I do *not* get this when using the full set of basis templates, so the fix might be on the side of the lightweight test data.  It would be great to get this fixed so that our tests are reproducibly robust (especially given how long the desisim tests take; it is pain to trip on this after 10 minutes of running...)

**Comment by moustakas** [comment]
When you run that snippet of code, the following warning is printed:
```
WARNING:templates.py:766:make_galaxy_templates: 1 spectra could not be computed given 
the input priors!
```
which affect the second model.  The reason this happens is because the second model gets assigned a z-band magnitude of ```20.482```, which is in the range, ```(19.0, 20.5)```, of the z-band magnitude priors hard-coded in ```desisim.templates.LRG```.  However, the LRG color and magnitude cuts coded in ```desitarget.cuts``` have a faint magnitude cut of ```z<20.46```, and so the second model fails to pass the flux cut.  

None of this is *wrong* per se, but it is a little annoying.  The easiest fix would be to just change the priors, although the default values are likely to get out of sync with the target selection cuts again.   Another possibility would be to add ```nocolorcuts=True``` in the tests when generating the templates, but that seems like a cheat.  The right thing is probably to test the template-generating code with ```nocolorcuts=True``` and then separately test the color-cut functionality.  I'll work on this but other suggestions are welcome.

**Comment by moustakas** [comment]
Should be addressed by #215.  Let's reopen if this issue appears again.
